overlord/servicestate/quota_control.go

```diff
@@ -44,6 +52,42 @@ func init() {
 	checkSystemdVersion()
 }
 
+func checkMemoryCGroupEnabled() {
+	memoryCGroupError = memoryCGroupEnabled()
+}
+
+var CGroupsFilePath = "/proc/cgroups"
+
+//	since the control groups can be enabled/disabled without the kernel config the only
+//	way to identify the status of memory control groups is via /proc/cgroups
+//	"cat /proc/cgroups | grep memory" returns the active status of memory control group
+//	and the 3rd parameter is the status
+//	0 => false => disabled 1 => true => enabled
+func memoryCGroupEnabled() error {
+	cgroupsFile, errFileOpen := os.Open(CGroupsFilePath)
+	if errFileOpen != nil {
+		return fmt.Errorf("internal error: cannot open cgroups file")
+	}
+	defer cgroupsFile.Close()
+	scanner := bufio.NewScanner(cgroupsFile)
+	scanner.Split(bufio.ScanLines)
+	for scanner.Scan() {
+		line := scanner.Text()
+		if strings.HasPrefix(line, "memory\t") {
+			isMemoryEnabled, errParseValue := strconv.ParseBool(strings.Fields(line)[3])
+			if errParseValue != nil {
+				return fmt.Errorf("cannot parse memory control group status")
+			}
```

Comment:
```suggestion
			isMemoryEnabled, ee := strconv.ParseBool(strings.Fields(line)[3])
			if err != nil {
				return fmt.Errorf("cannot parse memory control group status: %v", err)
			}
```
Also, the code should probably check that strings.Fileds() returns something that actually has at least 4 items.